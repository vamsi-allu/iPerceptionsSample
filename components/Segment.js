import React from 'react';
import {StyleSheet, View, Text} from 'react-native';

const Segment = (props) => {
  const data = props.data.item;
  return (
    <View style={styles.segment}>
      <Text ellipsizeMode="tail" numberOfLines={2} adjustsFontSizeToFit>
        {data.link}
      </Text>
      {props.children}
    </View>
  );
};

const styles = StyleSheet.create({
  segment: {
    width: '100%',
    paddingBottom: 35,
    paddingTop: 35,
    alignItems: 'center',
    justifyContent: 'center',
    flexDirection: 'row',
  },
});

export default Segment;
