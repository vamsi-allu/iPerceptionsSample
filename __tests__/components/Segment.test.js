import 'jsdom-global/register';
import React from 'react';
import Segment from '../../components/Segment';
import Enzyme, {shallow} from 'enzyme';
import toJson from 'enzyme-to-json';
import Adapter from 'enzyme-adapter-react-16';

Enzyme.configure({adapter: new Adapter()});

let props = {
  data: {
    item: {
      firstName: 'test',
      lasrtName: 'test',
    },
  },
};
describe('Segment Component', () => {
  it('renders correctly enzyme', () => {
    const wrapper = shallow(<Segment {...props} />);

    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('check the segment is loaded properly', () => {
    const component = shallow(<Segment {...props} />);

    expect(component.find('Text')).toHaveLength(1);
  });
});
