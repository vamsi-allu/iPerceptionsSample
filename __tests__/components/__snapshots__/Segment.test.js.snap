// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Segment Component renders correctly enzyme 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flexDirection": "row",
      "justifyContent": "center",
      "paddingBottom": 35,
      "paddingTop": 35,
      "width": "100%",
    }
  }
>
  <Text
    adjustsFontSizeToFit={true}
    ellipsizeMode="tail"
    numberOfLines={2}
  >
    test
     
  </Text>
</View>
`;
